src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -869,6 +877,15 @@ public bool CleanCacheAndDisposeIfUnused()
                 // time to determine whether a connection is too old and should be closed.
                 DateTimeOffset now = DateTimeOffset.UtcNow;
 
+                if (http2Connection != null)
+                {
+                    if (http2Connection.IsExpired(now, pooledConnectionLifetime, pooledConnectionIdleTimeout))
+                    {
+                        http2Connection.Dispose();
+                        InvalidateHttp2Connection(http2Connection);
```

Comment:
I don't think we need to call InvalidateHttp2Connection here. Just set _http2Connection to null. We already have the lock at this point, right?

Reply:
I was wondering about that but I wanted to call InvalidateHttp2Connection() just in case we add something more to it in future (logs, counters or other logic). I can change that if that is strong preference.

Reply:
I think we should change it. Taking a nested lock is usually a bad idea.